nmostesting/ControllerTest.py

```diff
@@ -294,7 +294,7 @@ def _populate_registry(self, test):
                 # Add sender to mock node
                 sender_json = self._create_sender_json(sender)
                 sender_ip_address = self.senders_ip_base + str(sender_ip_final_octet)
-                self.node.add_sender(sender_json, sender_ip_address)
+                self.node.add_sender(sender_json, sender_ip_address, sender.get("sdp_params"))
```

Comment:
Dictionary `get()` returns `None` by default if the key is absent, is that what you want?
Elsewhere you use `sender.get('sdp_params', {})`...

Reply:
Actually this illuminates an issue that confusingly there are two "types" of sender here - there is a sender definition in the ControllerTests (as specified in the test suites) that contains test specific metadata (description of the sender as it appears on the testing façade, whether sender is registered with mock registry etc.) and the sender definition that is used in the mock Node.  

- The sender definition in the test may or may not contain `sdp_params` - for instance the existing IS-04, IS-05 Controller tests do not have `sdp_params`.
- The sender definition in the Node always has `sdp_params` but they are None in the case of IS-04, IS-05 Controller tests.

Given the above is would probably be better  in ControllerTest to use `sender.get('sdp_params', {})`, with `{}` as the parameter default as the Node always expects `sdp_params`.